src/table-view/table-view.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import createReactClass from 'create-react-class'
+import PropTypes from 'prop-types'
+
+import { default as Table } from '../table'
+import { default as Select } from '../select'
+import { default as Button } from '../button'
+
+export default createReactClass({
+  displayName: 'TableView',
+  propTypes: {
+    rows: PropTypes.array,
+    columns: PropTypes.array,
+    filters: PropTypes.array,
+    onFilterChange: PropTypes.func,
+    onClearFilters: PropTypes.func
+  },
+  getDefaultProps() {
+    return {
+      rows: [],
+      columns: [],
+      filters: []
+    }
+  },
+
+  render() {
+    const { rows, columns, onFilterChange, onClearFilters } = this.props
+    const filters = sanitizeFilters(this.props.filters)
+
+    return (
+      <div style={{ display: 'flex', flexDirection: 'column' }}>
+        <div style={{ display: 'flex', width: '50%', alignItems: 'stretch' }}>
+          {filters.map(f => (
+            <div style={{ flex: '1', marginLeft: '10px' }}>
+              <Select
+                key={f.dataKey}
+                multiple
+                label={f.label}
+                value={f.value}
+                items={extractFilterValues(rows, f)}
```

Comment:
`items` in a `Select` filter have to be computed from the passed prop `rows`.

In order to avoid this recalculation, computed `items` for a filter need to be cached. We could do this by storing it in state, making it a derived-state.

However, `row` prop could change (say: number of rows increased in a re-render). This puts the derived-state on a flaky foundation, and is seen as an antipattern in React. Indeed, I have been bitten in past by derivedState. So I avoid it.

The official alternatives to avoid the repeated expensive computation are:

1. PureComponent - which implements `shouldComponentUpdate()` for us, but only performs shallow comparison in it. So its useful when the props and state are simple data structures. 
2. Hand coded `shouldComponentUpdate()` is not the most encouraged approach in react docs.
2. memoization - This is my preferred one. But I considered this optimization to be out of scope of this challenge.

Reference: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization. This section also provides an example quite similar to our usecase - a list with search filter.

Reply:
It's really good that you have thought about cases were the rows will change. 
I don't see were the anit-pattern would be here? correct me if I'm wrong but this looks to me a correct use case of derived-state, as the component state depends on props, written correctly it shouldn't cause problems.
anyway `createReactClass` doesn't have `getDerivedStateFromProps` so you'd probably need to use other life cycles

Reply:
> were the anit-pattern would be here
So my implication here was more it being a discouraged pattern than an antipattern.

> correct me if I'm wrong but this looks to me a correct use case of derived-state, as the component state depends on props, written correctly it shouldn't cause problems.
I agree. In fact this is the only valid usecase. However, within that usecase too, when given the choice, I prefer memoization for computed values over having to  manage the derived state and keeping it in sync with props. While react officially recommends the same, my personal experience also concurs.

In any case, I prefer to follow project convention above personal preference.